python/pylibcudf/pylibcudf/column.pyx

```diff
@@ -34,6 +34,22 @@ class _Ravelled:
         self.__cuda_array_interface__ = cai
 
 
+def _parse_array_interface(iface: dict) -> tuple[tuple, tuple | None, DataType]:
+    """
+    Parse and validate a CUDA or NumPy array interface dict.
+    """
+    typestr = iface["typestr"]
+    if typestr[0] == ">":
+        raise ValueError("Big-endian data is not supported")
+    elif typestr[0] not in ("<", "|"):
+        raise ValueError(f"Unsupported endianness prefix: {typestr[0]}")
+
+    dtype = _datatype_from_dtype_desc(typestr[1:])
+    shape = iface["shape"]
+    strides = iface.get("strides")
```

Comment:
*nit*: we could validate C-contiguous here.

Reply:
`from_gpumemoryview` is public and it needs the check.

Reply:
Done in https://github.com/rapidsai/cudf/pull/18425/commits/f361a5aa42991777cd4f461041b16acf62420287. I decided to make `from_gpumemoryview` private (ie `_from_gpumemoryview`). And I split out the parsing and validation into seperate functions.